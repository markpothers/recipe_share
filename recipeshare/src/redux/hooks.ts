import { TypedUseSelectorHook, useDispatch, useSelector } from "react-redux"
import type { AppDispatch } from "./store"
import type { RootStateType } from "./types"

export const useAppDispatch: () => AppDispatch = useDispatch
export const useAppSelector: TypedUseSelectorHook<RootStateType> = useSelector
