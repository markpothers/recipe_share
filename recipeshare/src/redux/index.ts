export { useAppDispatch, useAppSelector } from "./hooks";
export * from "./types";
export type { RootState } from "./store"
export {
	updateSingleRecipeList,
	updateAllRecipeLists,
	updateSingleChefList,
	updateAllChefLists,
	storeRecipeDetails,
	updateNewUserDetails,
	clearNewUserDetails,
	updateLoginUserDetails,
	clearLoginUserDetails,
	addRecipeLike,
	removeRecipeLike,
	addRecipeReShare,
	removeRecipeReShare,
	addRecipeMake,
	addMakePic,
	addMakePicChef,
	saveRemainingMakePics,
	updateComments,
	storeNewFollowers,
	updateLoggedInChef,
	storeChefDetails,
	clearChefDetails,
	stayLoggedIn
} from "./rootReducer"
export { default as rootReducer } from "./rootReducer"
