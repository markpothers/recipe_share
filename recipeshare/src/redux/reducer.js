const reducer = (currentState, action) => {
	switch (action.type) {
		case 'UPDATE_SINGLE_RECIPE_LIST':
			return { ...currentState, allRecipeLists: { ...currentState.allRecipeLists, [action.listKey]: action.recipeList}}
		case 'UPDATE_ALL_RECIPE_LISTS':
			return { ...currentState, allRecipeLists: action.allRecipeLists}
		case 'UPDATE_SINGLE_CHEF_LIST':
			return { ...currentState, allChefLists: { ...currentState.allChefLists, [action.listKey]: action.chefList}}
		case 'UPDATE_ALL_CHEF_LISTS':
			return { ...currentState, allChefLists: action.allChefLists}
		case 'STORE_RECIPE_DETAILS':
			return { ...currentState, recipe_details: action.recipe_details }
		case 'UPDATE_NEW_USER_DETAILS':
			return { ...currentState, newUserDetails: { ...currentState.newUserDetails, [action.parameter]: action.content } }
		case 'CLEAR_NEW_USER_DETAILS':
			// console.log("clearing new user details")
			return {
				...currentState, newUserDetails: {
					first_name: "",
					last_name: "",
					username: "",
					e_mail: "",
					password: "",
					password_confirmation: "",
					profile_text: "",
					country: "United States",
					image_url: ""
				}
			}
		case 'UPDATE_LOGIN_USER_DETAILS':
			return { ...currentState, loginUserDetails: { ...currentState.loginUserDetails, [action.parameter]: action.content } }
		case 'CLEAR_LOGIN_USER_DETAILS':
			// console.log("clearing login details")
			return {
				...currentState, loginUserDetails: {
					e_mail: "",
					password: "",
				}
			}
		case 'ADD_RECIPE_LIKE':
			return { ...currentState, recipe_details: { ...currentState.recipe_details, recipe_likes: currentState.recipe_details.recipe_likes + 1, likeable: false } }
		case 'UPDATE_COMMENTS':
			return { ...currentState, recipe_details: { ...currentState.recipe_details, comments: action.comments } }
		case 'REMOVE_RECIPE_LIKE':
			return { ...currentState, recipe_details: { ...currentState.recipe_details, recipe_likes: currentState.recipe_details.recipe_likes - 1, likeable: true } }
		case 'ADD_RECIPE_MAKE':
			return { ...currentState, recipe_details: { ...currentState.recipe_details, recipe_makes: currentState.recipe_details.recipe_makes + 1, makeable: false } }
		case 'ADD_RECIPE_SHARE':
			return { ...currentState, recipe_details: { ...currentState.recipe_details, recipe_shares: currentState.recipe_details.recipe_shares + 1, shareable: false } }
		case 'REMOVE_RECIPE_SHARE':
			return { ...currentState, recipe_details: { ...currentState.recipe_details, recipe_shares: currentState.recipe_details.recipe_shares - 1, shareable: true } }
		case 'ADD_MAKE_PIC':
			return { ...currentState, recipe_details: { ...currentState.recipe_details, make_pics: [action.makePic, ...currentState.recipe_details.make_pics] } }
		case 'ADD_MAKE_PIC_CHEF':
			return { ...currentState, recipe_details: { ...currentState.recipe_details, make_pics_chefs: [action.makePicChef, ...currentState.recipe_details.make_pics_chefs] } }
		case 'SAVE_REMAINING_MAKE_PICS':
			return { ...currentState, recipe_details: { ...currentState.recipe_details, make_pics: action.makePics } }
		case 'REMOVE_RECIPE_LIKES':
			return { ...currentState, recipes_details: { ...currentState.recipes_details, [action.listType]: { ...currentState.recipes_details[action.listType], recipe_likes: action.recipe_likes } } }
		case 'STORE_NEW_FOLLOWERS':
			return { ...currentState, chefs_details: { ...currentState.chefs_details, [action.chefID]: { ...currentState.chefs_details[action.chefID], followers: action.followers, chef_followed: !currentState.chefs_details[action.chefID].chef_followed } } }
		case 'LOG_IN_CHEF':
			// console.log("logging in chef")
			return {
				...currentState, loggedInChef: {
					id: action.id,
					username: action.username,
				}
			}
		case 'UPDATE_LOGGED_IN_CHEF':
			// console.log("updating logged in chef")
			return {
				...currentState, loggedInChef: {
					id: action.id,
					e_mail: action.e_mail,
					username: action.username,
					auth_token: action.auth_token,
					image_url: action.image_url,
					is_admin: action.is_admin
				}
			}
		case 'STORE_CHEF_LIST':
			return { ...currentState, chefs: { ...currentState.chefs, [action.chefType]: action.chefList } }
		case 'CLEAR_LISTED_CHEFS':
			return { ...currentState, chefs: { ...currentState.chefs, [action.chefType]: [] } }
		case 'APPEND_TO_CHEF_LISTS':
			// console.log(currentState.chefs[action.chefType])
			return { ...currentState, chefs: { ...currentState.chefs, [action.chefType]: [...currentState.chefs[action.chefType], ...action.chefList] } }
		case 'STORE_CHEF_DETAILS':
			// console.log(action.chef_details)
			return { ...currentState, chefs_details: { ...currentState.chefs_details, [action.chefID]: action.chef_details } }
		case 'CLEAR_CHEF_DETAILS':
			// console.log(action.chef_details)
			return { ...currentState, chef_details: {} }
		case 'TOGGLE_RECIPES_LIST_FILTER_CATEGORY':
			// console.log(action.chef_details)
			return { ...currentState, filter_settings: { ...currentState.filter_settings, [action.category]: action.value } }
		case 'CLEAR_RECIPES_LIST_FILTERS':
			return { ...currentState, filter_settings: action.clearedFilters, cuisine: "Any", serves: "Any" }
		case 'SET_RECIPES_LIST_CUISINE':
			// return { ...currentState, cuisine: action.cuisine }
			return { ...currentState, filterCuisines: { ...currentState.filterCuisines, [action.listChoice]: action.cuisine } }
		case 'SET_RECIPES_LIST_SERVES':
			return { ...currentState, serves: action.serves }
		case 'TOGGLE_NEW_RECIPE_FILTER_CATEGORY':
			// console.log(action.chef_details)
			return { ...currentState, newRecipeDetails: { ...currentState.newRecipeDetails, filter_settings: { ...currentState.newRecipeDetails.filter_settings, [action.category]: action.value } } }
		case 'CLEAR_NEW_RECIPE_FILTERS':
			return { ...currentState, newRecipeDetails: { ...currentState.newRecipeDetails, filter_settings: action.clearedFilters, cuisine: "Any", serves: "Any" } }
		case 'SET_NEW_RECIPE_CUISINE':
			return { ...currentState, newRecipeDetails: { ...currentState.newRecipeDetails, cuisine: action.cuisine } }
		case 'SET_NEW_RECIPE_SERVES':
			return { ...currentState, newRecipeDetails: { ...currentState.newRecipeDetails, serves: action.serves } }
		case 'STAY_LOGGED_IN':
			return { ...currentState, stayLoggedIn: action.value }
		default:
			return currentState
	}
}

export default reducer
