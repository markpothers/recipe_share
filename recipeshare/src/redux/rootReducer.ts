import { createSlice, PayloadAction } from "@reduxjs/toolkit";
import { ListChef, ListRecipe, LoginChef, MakePic, MakePicChef, Recipe, Comment } from "../centralTypes";
import { initialRootState } from "./initialRootState";
import {
	ParameterContent,
	RootStateType,
	StoreChefDetails,
	StoreNewFollowers,
	UpdateSingleChefList,
	UpdateSingleRecipeList,
} from "./types";

const rootReducer = createSlice({
	name: "root",
	initialState: initialRootState,
	reducers: {
		updateSingleRecipeList(state, action: PayloadAction<UpdateSingleRecipeList>) {
			state.allRecipeLists[action.payload.listKey] = action.payload.recipeList;
		},
		updateAllRecipeLists(state, action: PayloadAction<Record<string, ListRecipe[]>>) {
			state.allRecipeLists = action.payload;
		},
		updateSingleChefList(state, action: PayloadAction<UpdateSingleChefList>) {
			state.allChefLists[action.payload.listKey] = action.payload.chefList;
		},
		updateAllChefLists(state, action: PayloadAction<Record<string, ListChef[]>>) {
			state.allChefLists = action.payload;
		},
		storeRecipeDetails(state, action: PayloadAction<Recipe>) {
			state.recipe_details = action.payload;
		},
		updateNewUserDetails(state, action: PayloadAction<ParameterContent>) {
			state.newUserDetails[action.payload.parameter] = action.payload.content;
		},
		clearNewUserDetails(state) {
			state.newUserDetails = initialRootState.newUserDetails;
		},
		updateLoginUserDetails(state, action: PayloadAction<ParameterContent>) {
			state.loginUserDetails[action.payload.parameter] = action.payload.content;
		},
		clearLoginUserDetails(state) {
			state.loginUserDetails = initialRootState.loginUserDetails;
		},
		addRecipeLike(state) {
			state.recipe_details.recipe_likes += 1;
			state.recipe_details.likeable = false;
		},
		removeRecipeLike(state) {
			state.recipe_details.recipe_likes -= 1;
			state.recipe_details.likeable = true;
		},
		addRecipeReShare(state) {
			state.recipe_details.re_shares += 1;
			state.recipe_details.shareable = false;
		},
		removeRecipeReShare(state) {
			state.recipe_details.re_shares -= 1;
			state.recipe_details.shareable = true;
		},
		addRecipeMake(state) {
			state.recipe_details.recipe_makes += 1;
			state.recipe_details.makeable = false;
		},
		addMakePic(state, action: PayloadAction<MakePic>) {
			state.recipe_details.make_pics.unshift(action.payload);
		},
		addMakePicChef(state, action: PayloadAction<MakePicChef>) {
			state.recipe_details.make_pics_chefs.unshift(action.payload);
		},
		saveRemainingMakePics(state, action: PayloadAction<MakePic[]>) {
			state.recipe_details.make_pics = action.payload;
		},
		updateComments(state, action: PayloadAction<Comment[]>) {
			state.recipe_details.comments = action.payload;
		},
		storeNewFollowers(state, action: PayloadAction<StoreNewFollowers>) {
			state.chef_details[action.payload.chefID].followers = action.payload.followers;
			state.chef_details[action.payload.chefID].chef_followed =
				!state.chef_details[action.payload.chefID].chef_followed;
		},
		updateLoggedInChef(state, action: PayloadAction<RootStateType["loggedInChef"]>) {
			state.loggedInChef = {
				id: action.payload.id,
				e_mail: action.payload.e_mail,
				username: action.payload.username,
				auth_token: action.payload.auth_token,
				image_url: action.payload.image_url,
				is_admin: action.payload.is_admin,
				is_member: action.payload.is_member,
			};
		},
		storeChefDetails(state, action: PayloadAction<StoreChefDetails>) {
			state.chef_details[action.payload.chefID] = action.payload.chef_details;
		},
		clearChefDetails(state) {
			state.chef_details = initialRootState.chef_details;
		},
		stayLoggedIn(state, action: PayloadAction<boolean>) {
			state.stayLoggedIn = action.payload;
		},
	},
});

// dispatch(stayLoggedIn(true))

export const {
	updateSingleRecipeList,
	updateAllRecipeLists,
	updateSingleChefList,
	updateAllChefLists,
	storeRecipeDetails,
	updateNewUserDetails,
	clearNewUserDetails,
	updateLoginUserDetails,
	clearLoginUserDetails,
	addRecipeLike,
	removeRecipeLike,
	addRecipeReShare,
	removeRecipeReShare,
	addRecipeMake,
	addMakePic,
	addMakePicChef,
	saveRemainingMakePics,
	updateComments,
	storeNewFollowers,
	updateLoggedInChef,
	storeChefDetails,
	clearChefDetails,
	stayLoggedIn,
} = rootReducer.actions;

export default rootReducer.reducer;
