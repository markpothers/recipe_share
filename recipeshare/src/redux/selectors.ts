import { ReduxStateType } from "./types";
import { createSelector } from "@reduxjs/toolkit";

const selectRoot = (state: ReduxStateType) => state.root

export const getLoggedInChef = createSelector(selectRoot, root => root.loggedInChef)
export const getLoggedInUserDetails = createSelector(selectRoot, root => root.loginUserDetails)
export const getDeviceType = createSelector(selectRoot, root => root.deviceType)
export const getStayLoggedIn = createSelector(selectRoot, root => root.stayLoggedIn)
