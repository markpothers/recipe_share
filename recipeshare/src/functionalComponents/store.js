import { createStore, compose, applyMiddleware } from 'redux';
import ReduxThunk from 'redux-thunk'
import reducer from './reducer'

const initialState = {
  loggedInChef: {
    id: "",
    username: "",
    auth_token: "",
    imageURL: "",
    is_admin: false
  },
  recipes: {
    all: [],
    chef: [],
    chef_feed: [],
    chef_liked: [],
    chef_made: [],
    global_ranks: [],
    most_liked: [],
    most_made: []
  },
  recipes_details: {

  },
  newRecipeDetails: {
    name: "",
    instructions: ['test', 'test2'],
    ingredients: {
      ingredient1 :{
        name:"",
        quantity: "",
        unit: "Oz"
      }
    },
    difficulty: "0",
    time: "00:15",
    imageBase64: "",
    filter_settings: {
      "Breakfast": false,
      "Lunch": false,
      "Dinner": false,
      "Chicken": false,
      "Red meat": false,
      "Seafood": false,
      "Vegetarian": false,
      "Salad": false,
      "Vegan": false,
      "Soup": false,
      "Dessert": false,
      "Side": false,
      "Whole 30": false,
      "Paleo": false,
      "Freezer meal": false,
      "Keto": false,
      "Weeknight": false,
      "Weekend": false,
      "Gluten free": false,
      "Bread": false,
      "Dairy free": false,
      "White meat": false,
    },
    cuisine: "Any",
    serves: "Any",
    acknowledgement: ""
  },
  newUserDetails: {
    first_name: "",
    last_name: "",
    username: "",
    e_mail: "",
    password: "",
    password_confirmation: "",
    country: "United States",
    imageURL: "",
    profile_text: ""
  },
  loginUserDetails: {
    e_mail: "markpothers@hotmail.com",
    password: "123456"
  },
  chefs: {
    all_chefs: [],
    followed: [],
    most_liked_chefs: [],
    most_made_chefs: [],
    chef_followees: [],
    chef_followers: []
  },
  chefs_details: {},
  filter_settings: {
    "Breakfast": false,
    "Lunch": false,
    "Dinner": false,
    "Chicken": false,
    "Red meat": false,
    "Seafood": false,
    "Vegetarian": false,
    "Salad": false,
    "Vegan": false,
    "Soup": false,
    "Dessert": false,
    "Side": false,
    "Whole 30": false,
    "Paleo": false,
    "Freezer meal": false,
    "Keto": false,
    "Weeknight": false,
    "Weekend": false,
    "Gluten free": false,
    "Bread": false,
    "Dairy free": false,
    "White meat": false,
  },
  cuisine: "Any",
  serves: "Any",
  stayLoggedIn: false,
}

  const middleware = compose(
    applyMiddleware(ReduxThunk)
  )

  export const store = createStore(reducer, initialState, middleware)
