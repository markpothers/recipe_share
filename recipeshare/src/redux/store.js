import { createStore, compose, applyMiddleware } from 'redux';
import ReduxThunk from 'redux-thunk'
import reducer from './reducer'

// import { filters } from '../dataComponents/filters'

//exported ONLY for unit testing
export const initialState = {
	loggedInChef: {
		id: "",
		username: "",
		auth_token: "",
		image_url: "",
		is_admin: false
	},
	allRecipeLists: {},
	allChefLists: {},
	recipes_details: {

	},
	newRecipeDetails: {
		name: "",
		instructions: [
			'',
			'',
			'',
			'',
		],
		instructionImages: [],
		ingredients: {
			ingredient1: {
				name: "",
				quantity: "",
				unit: "Oz"
			}
		},
		difficulty: "0",
		time: "00:15",
		imageBase64: "",
		filter_settings: {
			"Breakfast": false,
			"Lunch": false,
			"Dinner": false,
			"Chicken": false,
			"Red meat": false,
			"Seafood": false,
			"Vegetarian": false,
			"Salad": false,
			"Vegan": false,
			"Soup": false,
			"Dessert": false,
			"Side": false,
			"Whole 30": false,
			"Paleo": false,
			"Freezer meal": false,
			"Keto": false,
			"Weeknight": false,
			"Weekend": false,
			"Gluten free": false,
			"Bread": false,
			"Dairy free": false,
			"White meat": false,
		},
		cuisine: "Any",
		serves: "Any",
		acknowledgement: ""
	},
	newUserDetails: {
		first_name: "",
		last_name: "",
		username: "",
		e_mail: "",
		password: "",
		password_confirmation: "",
		country: "United States",
		image_url: "",
		profile_text: ""
	},
	loginUserDetails: {
		e_mail: "",
		password: ""
	},
	chefs: {
		all_chefs: [],
		followed: [],
		most_liked_chefs: [],
		most_made_chefs: [],
		chef_followees: [],
		chef_followers: []
	},
	chefs_details: {},
	filter_settings: {
		"Breakfast": false,
		"Lunch": false,
		"Dinner": false,
		"Chicken": false,
		"Red meat": false,
		"Seafood": false,
		"Vegetarian": false,
		"Salad": false,
		"Vegan": false,
		"Soup": false,
		"Dessert": false,
		"Side": false,
		"Whole 30": false,
		"Paleo": false,
		"Freezer meal": false,
		"Keto": false,
		"Weeknight": false,
		"Weekend": false,
		"Gluten free": false,
		"Bread": false,
		"Dairy free": false,
		"White meat": false,
	},
	filterCuisines: {
		all: "Any",
		chef: "Any",
		chef_feed: "Any",
		chef_liked: "Any",
		chef_made: "Any",
		global_ranks: "Any",
		most_liked: "Any",
		most_made: "Any"
	},
	serves: "Any",
	stayLoggedIn: false,
}

export const middleware = compose( //exported ONLY for unit testing
	applyMiddleware(ReduxThunk)
)

export const store = createStore(reducer, initialState, middleware)
