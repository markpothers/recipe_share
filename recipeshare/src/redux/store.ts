import { createStore, compose, applyMiddleware } from "redux";
import ReduxThunk from "redux-thunk";
import reducer from "./reducer";
import { RootState } from "./types";

//exported ONLY for unit testing
export const initialState: RootState = {
	loggedInChef: {
		id: "",
		username: "",
		auth_token: "",
		image_url: "",
		is_admin: false,
	},
	allRecipeLists: {},
	allChefLists: {},
	recipe_details: {
		chef_id: 0,
		chef_username: "",
		comments: [],
		ingredient_uses: [],
		ingredients: [],
		instructions: [],
		instruction_images: [],
		likeable: true,
		make_pics: [],
		make_pics_chefs: [],
		makeable: true,
		re_shares: 0,
		recipe: {
			acknowledgement: null,
			acknowledgement_link: null,
			chef_id: 0,
			cook_time: 0,
			created_at: "",
			cuisine: "Any",
			description: "",
			difficulty: 0,
			hidden: false,
			id: 0,
			name: "",
			prep_time: 0,
			serves: "Any",
			show_blog_preview: false,
			time: 0,
			total_time: 0,
			updated_at: "",
			Breakfast: false,
			Lunch: false,
			Dinner: false,
			Chicken: false,
			"Red meat": false,
			Seafood: false,
			Vegetarian: false,
			Salad: false,
			Vegan: false,
			Soup: false,
			Dessert: false,
			Side: false,
			"Whole 30": false,
			Paleo: false,
			"Freezer meal": false,
			Keto: false,
			Weeknight: false,
			Weekend: false,
			"Gluten free": false,
			Bread: false,
			"Dairy free": false,
			"White meat": false,
		},
		recipe_images: [],
		recipe_likes: 0,
		recipe_makes: 0,
		shareable: true,
	},
	newRecipeDetails: {
		name: "",
		instructions: ["", "", "", ""],
		instructionImages: [],
		ingredients: {
			ingredient1: {
				name: "",
				quantity: "",
				unit: "Oz",
			},
		},
		difficulty: "0",
		time: "00:15",
		imageBase64: "",
		filter_settings: {
			Breakfast: false,
			Lunch: false,
			Dinner: false,
			Chicken: false,
			"Red meat": false,
			Seafood: false,
			Vegetarian: false,
			Salad: false,
			Vegan: false,
			Soup: false,
			Dessert: false,
			Side: false,
			"Whole 30": false,
			Paleo: false,
			"Freezer meal": false,
			Keto: false,
			Weeknight: false,
			Weekend: false,
			"Gluten free": false,
			Bread: false,
			"Dairy free": false,
			"White meat": false,
		},
		cuisine: "Any",
		serves: "Any",
		acknowledgement: "",
	},
	newUserDetails: {
		first_name: "",
		last_name: "",
		username: "",
		e_mail: "",
		password: "",
		password_confirmation: "",
		country: "United States",
		image_url: "",
		profile_text: "",
	},
	loginUserDetails: {
		e_mail: "",
		password: "",
	},
	chef_details: {
		chef: {
			country: "United States",
			created_at: "",
			id: 0,
			image_url: null,
			profile_text: "",
			username: "",
		},
		chef_commented: false,
		chef_followed: false,
		chef_liked: false,
		chef_made: false,
		chef_make_piced: false,
		chef_shared: false,
		comments: 0,
		comments_received: 0,
		followers: 0,
		following: 0,
		make_pics: 0,
		make_pics_received: 0,
		re_shares: 0,
		re_shares_received: 0,
		recipe_likes: 0,
		recipe_likes_received: 0,
		recipes: 0,
	},
	// filter_settings: {
	// 	Breakfast: false,
	// 	Lunch: false,
	// 	Dinner: false,
	// 	Chicken: false,
	// 	"Red meat": false,
	// 	Seafood: false,
	// 	Vegetarian: false,
	// 	Salad: false,
	// 	Vegan: false,
	// 	Soup: false,
	// 	Dessert: false,
	// 	Side: false,
	// 	"Whole 30": false,
	// 	Paleo: false,
	// 	"Freezer meal": false,
	// 	Keto: false,
	// 	Weeknight: false,
	// 	Weekend: false,
	// 	"Gluten free": false,
	// 	Bread: false,
	// 	"Dairy free": false,
	// 	"White meat": false,
	// },
	// filterCuisines: {
	// 	all: "Any",
	// 	chef: "Any",
	// 	chef_feed: "Any",
	// 	chef_liked: "Any",
	// 	chef_made: "Any",
	// 	global_ranks: "Any",
	// 	most_liked: "Any",
	// 	most_made: "Any"
	// },
	// serves: "Any",
	stayLoggedIn: false,
};

export const middleware = compose(
	//exported ONLY for unit testing
	applyMiddleware(ReduxThunk)
);

export const store = createStore(reducer, initialState, middleware);

export type AppDispatch = typeof store.dispatch;
