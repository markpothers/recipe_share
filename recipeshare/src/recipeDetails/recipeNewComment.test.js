import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { Provider } from 'react-redux'
import { store } from '../redux/store'

import RecipeNewComment from './recipeNewComment.js'
import { TouchableOpacity, TextInput } from 'react-native';

describe('RecipeNewComment', () => {

	let component
	let mockScrollToLocation
	let mockHandleCommentTextInput
	let mockSaveComment


	beforeAll(() => {
		// console.log('runs at the beginning of everything')
	})

	beforeEach(() => {
		// console.log('runs before every test')
		mockScrollToLocation = jest.fn()
		mockHandleCommentTextInput = jest.fn()
		mockSaveComment = jest.fn()
	})

	afterEach(() => {
		// console.log('runs after each test')
	})

	afterAll(() => {
		// console.log('runs after all tests have completed')
	})

	test('renders with some text and a confirm button', () => {
			act(() => {
				component = renderer.create(
					<RecipeNewComment 
					scrollToLocation={mockScrollToLocation} 
					username={"pothers"}
					image_url={"https://robohash.org/utautnemo.png?size=300x300&set=set1"}
					commentText={"Here is my test comment text"} 
					handleCommentTextInput={mockHandleCommentTextInput} 
					saveComment={mockSaveComment}
					/>
				)
			})
			const image = component.toJSON()
			expect(image).toMatchSnapshot()
			let root = component.root
			const input = root.findAllByType(TextInput)
			expect(input.length).toEqual(1)
	})

});
