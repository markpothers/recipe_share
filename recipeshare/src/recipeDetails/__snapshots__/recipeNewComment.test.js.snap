// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeNewComment renders with some text and a confirm button 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#104e01",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "marginLeft": "2%",
      "marginRight": "2%",
      "marginTop": 3,
      "paddingBottom": "1%",
      "width": "96%",
    }
  }
>
  <View
    style={
      Object {
        "flex": 2,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://robohash.org/utautnemo.png?size=300x300&set=set1",
        }
      }
      style={
        Object {
          "borderRadius": 5,
          "height": 60,
          "marginRight": "4%",
          "width": "96%",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 8,
      }
    }
  >
    <Text
      maxFontSizeMultiplier={2}
      style={
        Object {
          "color": "#505050",
          "fontSize": 33.65552620959067,
          "fontStyle": "italic",
          "marginLeft": "3%",
          "marginRight": "3%",
        }
      }
    >
      pothers
      :
    </Text>
    <TextInput
      allowFontScaling={true}
      autoFocus={true}
      maxFontSizeMultiplier={2}
      multiline={true}
      numberOfLines={2}
      onChangeText={[Function]}
      placeholder="Type comment here..."
      rejectResponderTermination={true}
      style={
        Object {
          "color": "#505050",
          "fontSize": 33.65552620959067,
          "marginLeft": "3%",
          "marginRight": "3%",
          "paddingLeft": 15,
        }
      }
      underlineColorAndroid="transparent"
      value="Here is my test comment text"
    />
  </View>
</View>
`;
